### PR TITLE
Implement random packet dropping

### PR DESCRIPTION
**Pull request by serathius**
cc

### CODE REVIEW

**Comment by serathius** [comment]
cc @ahrtr @fuweid

**Review by fuweid — APPROVED** [review]
LGTM

File: tests/robustness/failpoint/network.go [hunk 1]
```diff
@@ -167,9 +168,43 @@ func (f delayPeerNetworkFailpoint) Inject(ctx context.Context, t *testing.T, lg
...
+
+func (f dropPeerNetworkFailpoint) Inject(ctx context.Context, t *testing.T, lg *zap.Logger, clus *e2e.EtcdProcessCluster) error {
+	member := clus.Procs[rand.Int()%len(clus.Procs)]
+	proxy := member.PeerProxy()
+
+	proxy.ModifyRx(f.modifyPacket)
+	proxy.ModifyTx(f.modifyPacket)
+	lg.Info("Dropping traffic from and to member", zap.String("member", member.Config().Name), zap.Int("probability", f.dropProbabilityPercent))
+	time.Sleep(f.duration)
+	lg.Info("Traffic delay removed", zap.String("member", member.Config().Name))
```
Comment (chaochn47) [inline tests/robustness/failpoint/network.go:191]:
```suggestion
	lg.Info("Traffic drop removed", zap.String("member", member.Config().Name))
```

**Review by chaochn47 — APPROVED** [review]

File: tests/robustness/failpoint/network.go (hunk 1 above)
Comment (serathius) [reply tests/robustness/failpoint/network.go:191]: Done

**Review by ahrtr — APPROVED** [review]

**Pull request merged**